cw_decred/lib/wallet.dart

```diff
@@ -65,63 +82,318 @@ abstract class DecredWalletBase extends WalletBase<DecredBalance,
 
   Future<void> init() async {
     updateBalance();
-    // TODO: update other wallet properties such as syncStatus, walletAddresses
-    // and transactionHistory with data from libdcrwallet.
   }
 
+  void performBackgroundTasks() async {
+    if (!checkSync()) {
+      return;
+    }
+    var checkTxs = this.checkBalance;
+    if (!checkTxs) {
+      final res = libdcrwallet.bestBlock(walletInfo.name);
+      final decoded = json.decode(res);
+      final hash = decoded["hash"] ?? "";
+      if (this.bestHash != hash) {
+        this.bestHash = hash;
+        this.bestHeight = decoded["height"] ?? "";
+        checkTxs = true;
+      }
+    }
+    if (!checkTxs) {
+      return;
+    }
+    this.checkBalance = false;
+    updateBalance();
+    var from = 0;
+    while (true) {
+      // Transactions are returned from newest to oldest. Loop fetching 5 txn
+      // at a time until we find a batch with txn that no longer need to be
+      // updated.
+      final txs = await this.fetchNTransactions(from);
+      if (txs.length == 0) {
+        return;
+      }
+      if (this.transactionHistory.update(txs)) {
+        return;
+      }
+      from += 5;
+    }
+  }
+
+  bool checkSync() {
+    final syncStatusJSON = libdcrwallet.syncStatus(walletInfo.name);
+    final decoded = json.decode(syncStatusJSON);
+
+    final syncStatusCode = decoded["syncstatuscode"] ?? 0;
+    final syncStatusStr = decoded["syncstatus"] ?? "";
+    final targetHeight = decoded["targetheight"] ?? 1;
+    final numPeers = decoded["numpeers"] ?? 0;
+    // final cFiltersHeight = decoded["cfiltersheight"] ?? 0;
```

Comment:
why is this commented?  not needed currently?

Reply:
Yeah not using it below. I'll try to remember why.